.github/workflows/release.yml

```diff
@@ -0,0 +1,154 @@
+name: Release build
+env:
+  DOCKER_REGISTRY: "localhost:5000"
+  TAG: "latest"
+  DOCKER_ORG: enmasse
+  KUBERNETES_NAMESPACE: "enmasse-infra"
+  SYSTEMTEST_PROFILE: smoke
+  EXTERNAL_REGISTRY: "quay.io"
+  DOCKER_USER: "obabec+actions"
+
+on:
+  push:
+    branches:
+      - master
+      - actions
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Dump GitHub context
+      env:
+        GITHUB_CONTEXT: ${{ toJson(github) }}
+      run: echo "$GITHUB_CONTEXT"
+    - name: Export version
+      id: version
+      run: | 
+        echo "::set-env name=VERSION::$(grep "release.version" pom.properties| cut -d'=' -f2)"
+
+    - name: Install maven
+      run: sudo apt install maven
+
+    - name: Setup NodeJS
+      run: |
+        sudo apt-get install curl
+        curl -sL https://deb.nodesource.com/setup_13.x
+        sudo -E bash -
+        sudo apt-get install nodejs
+
+    - name: Install golang
+      run: |
+        sudo add-apt-repository ppa:longsleep/golang-backports
+        sudo apt-get update
+        sudo apt-get install golang-1.12
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11.0.2
+
+    - name: setup-docker
+      uses: docker-practice/actions-setup-docker@0.0.1
+    
+    - name: Setup Ruby environment with ruby-build
+      uses: clupprich/ruby-build-action@v1.0.0
+  
+    - name: Install maven
+      run: sudo apt install maven
+
+    - name: Install asciidoctor
+      run: gem install asciidoctor
+
+    - name: Minikube setup with registry
+      uses: obabec/minikube-setup-action@v0.0.4-alpha
+      with:
+        k8s-version: 1.15.0
+
+    - name: Make images
+      run: make
+
+    - name: Collect logs from Unit tests
+      run: ./.github/scripts/collectLogs.sh
+      
+    - name: Build images
+      id: build
+      run: make docker_build
+      
+    - name: Tag images
+      run: make docker_tag
+      
+    - name: Push images
+      run: make -j 4 docker_tag docker_push
+      
+    - name: Generate templates
+      run: make templates
+
+    - name: Smoke tests
+      run: ./.travis/test.sh        
+
+    - name: Collect logs
+      run: ./.github/scripts/collectLogs.sh
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v1.0.0
+      with:
+        name: test-logs
+        path: artifacts
+
+    - name: Change docker reg
+      run: |
+        echo "::set-env name=DOCKER_REGISTRY::quay.io"
+
+    - name: Build clean images
+      if: success()
+      run: |
+        make clean
+        make SKIP_TESTS=true
+        make docker_build
+        make docker_tag
+        make templates
```

Comment:
I think that we should have the images pushed before the test (I know I may have stated otherwise, but this just occured to me). Otherwise, I'm worried that we're releasing templates we have not really tested. Its better to push images prematurely and have tests fail, than to push bad templates without knowing it.

Reply:
I may just tag images from local registry to quay instead of building new.

Reply:
The problem is with the templates, as they would have to be regenerated. IMO we should not regenerate them after testing.

Reply:
Ye I should skip whole this sections and just re-tag from localhost:*** to quay.io***. Is this acceptable ?